lib/components/normal-components/Fiducial.ts

```diff
@@ -0,0 +1,75 @@
+import { NormalComponent } from "lib/components/base-components/NormalComponent/NormalComponent"
+import {
+  distance,
+  type PcbSmtPadCircle,
+  type SourceSimpleFiducial,
+} from "circuit-json"
+import { commonComponentProps } from "@tscircuit/props"
+import { FTYPE } from "lib/utils/constants"
+
+export const fiducialProps = commonComponentProps.extend({
+  soldermaskPullback: distance.optional(),
+  padDiameter: distance.optional(),
+})
+
+export class Fiducial extends NormalComponent<typeof fiducialProps> {
+  ftype: "simple_fiducial" = "simple_fiducial"
+
+  get config() {
+    return {
+      componentName: "Fiducial",
+      zodProps: fiducialProps,
+      sourceFtype: FTYPE.simple_fiducial,
+      shouldRenderAsSchematicBox: false,
+    }
+  }
+
+  doInitialPcbComponentRender(): void {
+    if (this.root?.pcbDisabled) return
+    const { db } = this.root!
+    const { _parsedProps: props } = this
+    const { pcbX, pcbY } = this.getResolvedPcbPositionProp()
+
+    const padDiameter = props.padDiameter ?? 1
+    const pcb_component = db.pcb_component.insert({
+      center: { x: pcbX, y: pcbY },
+      width: padDiameter,
+      height: padDiameter,
+      layer: props.layer ?? "top",
+      rotation: props.pcbRotation ?? 0,
+      source_component_id: this.source_component_id!,
+      subcircuit_id: this.getSubcircuit().subcircuit_id ?? undefined,
+      do_not_place: props.doNotPlace ?? false,
+      obstructs_within_bounds: false,
+    })
+
+    this.pcb_component_id = pcb_component.pcb_component_id
+
+    const soldermaskPullback = props.soldermaskPullback ?? padDiameter / 2
+
+    db.pcb_smtpad.insert({
+      shape: "circle",
+      x: pcbX,
+      y: pcbY,
+      radius: padDiameter / 2,
+      layer: props.layer ?? "top",
+      pcb_component_id: this.pcb_component_id,
+      subcircuit_id: this.getSubcircuit().subcircuit_id ?? undefined,
+      is_covered_with_solder_mask: true,
+      soldermask_margin: soldermaskPullback,
+    } as PcbSmtPadCircle)
+  }
+
+  doInitialSourceRender() {
+    const { db } = this.root!
+    const source_component = db.source_component.insert({
+      ftype: "simple_fiducial",
+      name: this.name,
+    } as Omit<SourceSimpleFiducial, "source_component_id" | "type">)
+    this.source_component_id = source_component.source_component_id
+  }
+
+  doInitialSchematicComponentRender() {
```

Comment:
just set `noSchematicRepresentation` on this- (don't set a prop- props are not allowed to be touched, it depends on how the implementation works but something like this...

```
get hasSchematicRepresenation() {
  return !this._parsedProps.noSchematicRepresentation
}
```

in NormalComponent then an override in `Fiducial` is a good idea

Reply:
it's also possible @ShiboSoftwareDev that this is not a NormalComponent, this may be a PrimitiveComponent, in which case you don't really have to do this.